### ISSUE TITLE
.dependsOn(subproject % "jooq") doesn't work

### ISSUE THREAD

**Issue by tekacs** [kxbmap/sbt-jooq#1]
In order to provide a CodegenStrategy (the jOOQ kind, not the one defined in this plugin), it's necessary to compile some other code before running the jOOQ code generator.

In our project, the way we do this is by depending on another package, which contains the relevant code (a class to be used by the jOOQ code generator).

Unfortunately, depending on this project from the jOOQ project in the jooq scope doesn't seem to work (everything but that works fine).

Example:

```scala
lazy val upstream = project

lazy val jooq =
  project
  .enablePlugins(JooqCodegen)
  .settings(
    libraryDependencies := "org.postgresql" % "postgresql" % "version", // this works fine
    // other jooq settings 
    jooqCodegenConfig := <generator><strategy>misc.JooqStrategy</strategy></generator> // snipped
  )
  .dependsOn(upstream % "jooq") // this doesn't work
```

The error is at the bottom of this post.

Trying with `.dependsOn(upstream % "jooqCodegen")` instead fails with `Error: Could not find or load main class org.jooq.util.GenerationTool`.

Any thoughts on why this might be, or a solution for providing a strategy another way? Thanks!

---

It fails with the following (seemingly simply not selecting the dependency):

```
> project jooq
[info] Set current project to jooq (in build file:/C:/Users/<snip>/)
> compile
Mar 14, 2017 6:44:12 PM org.jooq.tools.JooqLogger info
INFO: Initialising properties  : C:\Users\tekacs\AppData\Local\Temp\jooq-codegen-5419422408100924924.xml
Mar 14, 2017 6:44:12 PM org.jooq.tools.JooqLogger error
SEVERE: Cannot read C:\Users\tekacs\AppData\Local\Temp\jooq-codegen-5419422408100924924.xml. Error : misc.JooqStrategy
java.lang.ClassNotFoundException: misc.JooqStrategy
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at org.jooq.util.GenerationTool.loadClass(GenerationTool.java:672)
        at org.jooq.util.GenerationTool.run(GenerationTool.java:284)
        at org.jooq.util.GenerationTool.generate(GenerationTool.java:199)
        at org.jooq.util.GenerationTool.main(GenerationTool.java:171)

[trace] Stack trace suppressed: run last jooq/*:jooqCodegen for the full output.
[error] (jooq/*:jooqCodegen) jOOQ codegen failure: -1
[error] Total time: 1 s, completed 14-Mar-2017 18:44:12
```

**Comment by tekacs** [comment]
The solution to this, for anyone else who comes up against this, is as follows:

```
lazy val upstream = project

lazy val jooq =
  project
  .enablePlugins(JooqCodegen)
  .settings(
    libraryDependencies := "org.postgresql" % "postgresql" % "version", // this works fine
    // other jooq settings 
    jooqCodegenConfig := <generator><strategy>upstream.JooqStrategy</strategy></generator>, // snipped
    (managedClasspath in jooq) ++= (fullClasspath in (upstream, Compile)).value
  )
```

I discovered the above by noting that [sbt-jooq uses the classpath of the jooq config's update task](https://github.com/kxbmap/sbt-jooq/blob/8d19df879db5d297446643ef603dc78c91dd1c68/src/main/scala/com/github/kxbmap/sbt/jooq/JooqCodegen.scala#L54) as its runtime managedClasspath and ignores other things you might introduce to the `jooq` scope (such as dependencies, unmanagedClasspath, etc.). So the solution for now is to manually push a (tracked) task onto the classpath that it's reading from.

I'll leave this issue open for now, since this should probably be documented or the behaviour changed.

**Comment by kxbmap** [comment]
Thank you for reporting. I'll look into the issue.

**Comment by kxbmap** [comment]
Fixed by #3 and published version 0.3.0 to Maven Central.
